### ISSUE TITLE
Extend API to allow removal of eduGAIN entities from AAF feed

### ISSUE THREAD

**Issue by rianniello** [ausaccessfed/saml-service#265]
I had to manually play with SAML service prod data to remove an edugain SP from our AAF metadata feed.

This API https://github.com/ausaccessfed/saml-service/pull/254 needs to be extended to do this. The associated Slab article needs to be updated too.

Code I used

``` ruby 
> ed = EntityId[uri: "https://shibboleth2sp.sams2.oup.com/shibboleth"].raw_entity_descriptor
> ed.known_entity.tags.each { |t| puts "#{t.name} #{t.derived}" }
edugain false
aaf false
sp false
aaf-edugain false
> ed.known_entity.untag_as('aaf')
=> []
> ed.save(raise_on_save_failure: true)
=> #<RawEntityDescriptor @values={...
```

I am not confident for full tag control via API just yet, though this is a step towards it.

These tags basically control everything in SAML Service. By having the `aaf` tag, the SP ends up in our AAF metadata feed: https://md.aaf.edu.au/aaf-metadata.xml

**Comment by vladimir-mencl-eresearch** [comment]
Hi @rianniello ,

This looks interesting.  How did the SP get the `aaf` tag in the first place?

If it just "arrived" from eduGAIN and was approved via the API, it would have received the `aaf-edugain-verified` tag.

I'd agree it would make sense to have API able to do the "inverse" of the current actions - so stop export of an IdP/SP  into eduGAIN, and also stop import of an IdP/SP from eduGAIN - either an SP explicitly approved via the API, or an IdP/SP imported implicitly.  In the last case, it would go beyond reversing the action of an SP.

I'm on the fence in regard to a generic tag manipulation API (not sure whether needed / what havoc it could cause, vs. it might be generally useful).

But I'm surprised the SP actually ended with the `aaf` tag.  Hmm, wondering, might be the case of an SP known independently both to eduGAIN and to AAF?

But even without generic tag manipulation API, the use cases above - reversing API action, or a generic import block, would be useful.

Cheers,
Vlad

**Comment by rianniello** [comment]
> This looks interesting. How did the SP get the aaf tag in the first place?
> If it just "arrived" from eduGAIN and was approved via the API, it would have received the aaf-edugain-verified tag.

Agree. I struggled to understand what happened with entity. Might've been done by hand.

> I'm on the fence in regard to a generic tag manipulation API (not sure whether needed / what havoc it could cause, vs. it might be generally useful).

Agree, perhaps too much of a shotgun at this stage. I like idea of letting use cases emerge then turning them into API calls.

**Comment by rianniello** [comment]
Perhaps this ticket requires some deeper understanding of SAML service, though just starting with reversing the current API calls is a good starting point.

**Comment by vladimir-mencl-eresearch** [comment]
I'd also suggest adding the blocking of default imports (IdP/SP) - this might be quite handy in case of issues caused by automatic import.
I have seen MDUI tags with unusual language codes (3-lettered) break XML parsing on "legacy platforms" (RHEL 5, about 5 years ago).  Being able to quickly blacklist an entity from import (via API) might be handy.
